tests/cli/test_build_and_deploy_aws.sh

```diff
@@ -0,0 +1,187 @@
+#!/bin/bash
+# Note: execute this file from the project root directory
+
+#####
+#
+# Make sure we can build an image and deploy it inside AWS!
+#
+#####
+
+. /usr/share/beakerlib/beakerlib.sh
+
+CLI="./src/bin/composer-cli"
+
+
+rlJournalStart
+    rlPhaseStartSetup
+        if [ -z "$AWS_ACCESS_KEY_ID" ]; then
+            rlFail "AWS_ACCESS_KEY_ID is empty!"
+        else
+            rlLogInfo "AWS_ACCESS_KEY_ID is configured"
+        fi
+
+        if [ -z "$AWS_SECRET_ACCESS_KEY" ]; then
+            rlFail "AWS_SECRET_ACCESS_KEY is empty!"
+        else
+            rlLogInfo "AWS_SECRET_ACCESS_KEY is configured"
+        fi
+
+        AWS_BUCKET="${AWS_BUCKET:-composerredhat}"
+        AWS_REGION="${AWS_REGION:-us-east-1}"
+
+        rlLogInfo "AWS_BUCKET=$AWS_BUCKET"
+        rlLogInfo "AWS_REGION=$AWS_REGION"
+
+        if ! rlCheckRpm "python3-pip"; then
+            rlRun -t -c "dnf -y install python3-pip"
+            rlAssertRpm python3-pip
+        fi
+
+        rlRun -t -c "pip3 install awscli"
+
+        # aws configure
+        mkdir ~/.aws/
+
+        cat > ~/.aws/config << __EOF__
```

Comment:
I think we need to make sure that tests can either be run as a user, without destroying the user's environment, or add protections to stop the test if it isn't running inside a test environment like docker, podman, etc. If I accidentally ran this locally I'd be fairly well annoyed that my personal aws credentials had been wiped.
(And this may not be the only case of this happening, need to look at the rest of the tests)

Reply:
Updated to not overwrite the files if they exist